sdk/eventhub/azure-eventhub/azure/eventhub/_client_base.py

```diff
@@ -226,14 +230,31 @@ def _management_request(self, mgmt_msg, op_type):
                     mgmt_msg,
                     constants.READ_OPERATION,
                     op_type=op_type,
-                    status_code_field=b"status-code",
-                    description_fields=b"status-description",
+                    status_code_field=MGMT_STATUS_CODE,
+                    description_fields=MGMT_STATUS_DESC,
                 )
-                status_code = response.application_properties[b"status-code"]
+                status_code = response.application_properties[MGMT_STATUS_CODE]
                 if status_code < 400:
                     return response
-                raise errors.AuthenticationException(
-                    "Management request error. Status code: {}".format(status_code)
+                if status_code in [401]:
+                    raise AuthenticationError(
```

Comment:
Is it better to raise an uamqp exception here so the "except" clause can handle it and retry?
Finally it will be wrapped into a EventHubError or its subclass to users

Reply:
Aah interesting - so `_handle_exception` actually _raises_ an EventHubError, and only _returns for retry_ if it's a uamqp error.
I missed that subtlety - I will adjust.